File: backend/src/main/java/org/cms/Enums/Role.java
package org.cms.Enums;

import java.util.HashMap;
import java.util.Map;

public enum Role {
    ADMIN(0),
    DOCTOR(1),
    RECEPTIONIST(2),
    PHARMACIST(3);

    private final int value;
    private static final Map<Integer, Role> map = new HashMap<>();

    private Role(int value) {
        this.value = value;
    }

    static {
        for(Role role : Role.values())
            map.put(role.value, role);
    }

    public static Role valueOf(int value){
        return map.get(value);
    }
}

File: frontend/src/enums/Role.ts
export enum Role {
    ADMIN = 0,
    DOCTOR = 1,
    RECEPTIONIST = 2,
    PHARMACIST = 3
};

export namespace Role {
    export function valueOf(value: string | number): Role | ((value: string | number) => Role | typeof valueOf | undefined) | undefined {
        if(typeof value === 'string'){
            if(!(value in Role)) return undefined;

            return Role[value as keyof typeof Role];
        }

        if(typeof value === 'number'){
            return Object.values(Role).includes(value as Role) ? value as Role : undefined;
        }
    }
};